pkg/gadgets/execsnoop/tracer/core/tracer.go

```diff
@@ -0,0 +1,208 @@
+//go:build linux
+// +build linux
+
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+// #include <linux/types.h>
+// #include "./bpf/execsnoop.h"
+import "C"
+
+import (
+	"fmt"
+	"path/filepath"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+	"github.com/cilium/ebpf/perf"
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/execsnoop/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+//go:generate sh -c "GOOS=$(go env GOHOSTOS) GOARCH=$(go env GOHOSTARCH) go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang execsnoop ./bpf/execsnoop.bpf.c -- -I./bpf/ -I../../../../ -target bpf -D__TARGET_ARCH_x86"
+
+type Config struct {
+	// TODO: Make it a *ebpf.Map once
+	// https://github.com/cilium/ebpf/issues/515 and
+	// https://github.com/cilium/ebpf/issues/517 are fixed
+	MountnsMap string
+}
+
+type Tracer struct {
+	c             *Config
+	objs          execsnoopObjects
+	enterLink     link.Link
+	exitLink      link.Link
+	resolver      containercollection.ContainerResolver
+	eventCallback func(types.Event)
+	done          chan bool
```

Comment:
I think this channel is not being used here

Reply:
Removed it.